TSPL.docc/ReferenceManual/Attributes.md

```diff
@@ -210,8 +210,77 @@ including important milestones.
   obsoleted: <#version number#>
   ```
   The *version number* consists of one to three positive integers, separated by periods.
+
+- The `noasync` argument indicates that the declared symbol can't be used directly
+  in an asynchronous context.
+
+  Because Swift concurrency can resume on a different thread
+  after a potential suspension point,
+  using elements like thread-local storage, locks, mutexes, or semaphores
+  across suspension points can lead to incorrect results.
+
+  To avoid any errors, add an `@available(*, noasync)` attribute to the symbol's declaration:
+
+  ```swift
+  extension pthread_mutex_t {
+    
+    @available(*, noasync)
+    mutating func lock() {
+        pthread_mutex_lock(&self)
+    }
+    
+    @available(*, noasync)
+    mutating func unlock() {
+        pthread_mutex_unlock(&self)
+    }
+  }
+  ```
+ 
+  This attribute tells the compiler to raise a build error 
+  when the symbol is used in an asynchronous context. 
+  You can also use the `message` argument to provide additional information
+  about the symbol.
+
+  ```swift
+  @available(*, noasync, message: "Migrate locks to Swift concurrency.")
+  mutating func lock() {
+    pthread_mutex_lock(&self)
+  }
+  ```
+
+  If you can guarantee that your code uses a potentially unsafe symbol in a safe manner, 
+  You can wrap it in a synchronous function and call that function 
```

Comment:
Typo "You" mid-sentence.  Probably don't need "you can" anyhow though.

```suggestion
  wrap it in a synchronous function and call that function 
```

Maybe "that wrapper function" instead of "that function", for clarity.